modules/calib/src/calibration.cpp

```diff
@@ -1182,6 +1182,427 @@ static double stereoCalibrateImpl(
+                   (points.rows == 1 && points.cols == pointsTotal && cn == 2)));
+
+        A[k] = Matx33d::eye();
+
+        points.convertTo(imagePoints[k], CV_64F);
+        imagePoints[k] = imagePoints[k].reshape(2, 1);
+        
+        cameraMatrix.convertTo(A[k], CV_64F);
+        Mat tdist( distCoeffs.size(), CV_MAKETYPE(CV_64F, distCoeffs.channels()), distInitial[k].val);
+        distCoeffs.convertTo(tdist, CV_64F);
+
+    }
+
+    // TODO: Not suppporting same focal length and fixed aspect ratio
+    // if( flags & CALIB_SAME_FOCAL_LENGTH )
+    // {
+    //     A[0](0, 0) = A[1](0, 0) = (A[0](0, 0) + A[1](0, 0))*0.5;
+    //     A[0](0, 2) = A[1](0, 2) = (A[0](0, 2) + A[1](0, 2))*0.5;
+    //     A[0](1, 1) = A[1](1, 1) = (A[0](1, 1) + A[1](1, 1))*0.5;
+    //     A[0](1, 2) = A[1](1, 2) = (A[0](1, 2) + A[1](1, 2))*0.5;
+    // }
+
+    // if( flags & CALIB_FIX_ASPECT_RATIO )
+    // {
+    //     for(int k = 0; k < 2; k++ )
+    //         aspectRatio[k] = A[k](0, 0) / A[k](1, 1);
+    // }
+
+    // we optimize for the inter-camera R(3),t(3)
+    // Param mapping is:
+    // - from 0 next 6: stereo pair Rt, from 6+i*6 next 6: Rt for each ith camera of nimages,
+    nparams = 6*(nimages+1);
+
+    std::vector<uchar> mask(nparams, (uchar)1);
+
+    std::vector<double> param(nparams, 0.);
+
+    // storage for initial [om(R){i}|t{i}] (in order to compute the median for each component)
+    std::vector<double> rtsort(nimages*6);
+    /*
+       Compute initial estimate of pose
+       For each image, compute:
+          R(om) is the rotation matrix of om
+          om(R) is the rotation vector of R
+          R_ref = R(om_right) * R(om_left)'
+          T_ref_list = [T_ref_list; T_right - R_ref * T_left]
+          om_ref_list = {om_ref_list; om(R_ref)]
+       om = median(om_ref_list)
+       T = median(T_ref_list)
+    */
+    int pos = 0;
+    for(int i = 0; i < nimages; i++ )
+    {
+        int ni = _npoints.at<int>(i);
+        Mat objpt_i = objectPoints.colRange(pos, pos + ni);
+        Matx33d R[2];
+        Vec3d rv, T[2];
+        for(int k = 0; k < 2; k++ )
+        {
+            Mat imgpt_ik = imagePoints[k].colRange(pos, pos + ni);
+            if (!isFisheyes[k])
+                solvePnP(objpt_i, imgpt_ik, A[k], distInitial[k], rv, T[k], false, SOLVEPNP_ITERATIVE );
+            else {
+                // TODO: to the initialization of camer matrix for the fisheye camera model, should take better choise than solvePnP
+                Mat& cameraMatrix = (k == 0) ? _cameraMatrix1 : _cameraMatrix2; 
+                Mat& distCoeffs = (k == 0) ? _distCoeffs1 : _distCoeffs2; 
+                Mat imagePointsNormalized;
+                cv::fisheye::undistortPoints(imgpt_ik, imagePointsNormalized, cameraMatrix, distCoeffs, noArray(), cameraMatrix);
+                solvePnP(objpt_i, imagePointsNormalized, cameraMatrix, noArray(), rv, T[k], false, SOLVEPNP_ITERATIVE);
```

Comment:
@ivashmak @lpanaf I propose to add fisheye overload for solvePnP to namespace `fisheye`.

Reply:
Do you want to make a separate function within `fisheye` namespace that takes as input distorted points from fisheye camera, internally undistorts them and runs the PnP solver?

Reply:
In general - yes. We have solvePnP that expects pinhole model. Let's introduce fisheye::solvePnP that expects fisheye camera model internally. The first implementation may be done with just undistort->solvePnP, if it produces reasonable result.

Reply:
Yes, that makes sense. It looks like just a combination of two functions. @lpanaf will you do it since it is your pull request, or I can try to push here too?

Reply:
I can add it, no problem :)